sdk/core/azure-core/inc/context.hpp

```diff
@@ -144,19 +144,30 @@ namespace Azure { namespace Core {
   private:
     struct ContextSharedState
     {
-      std::shared_ptr<ContextSharedState> Parent;
-      time_point CancelAt;
-      std::string Key;
-      ContextValue Value;
+      std::shared_ptr<ContextSharedState> const Parent;
+      std::chrono::milliseconds CancelAtMsecSinceEpoch;
+      std::string const Key;
+      ContextValue const Value;
 
-      explicit ContextSharedState() : CancelAt(time_point::max()) {}
+      static constexpr std::chrono::milliseconds ToMsecSinceEpoch(time_point time)
```

Comment:
This is done to explicitly store milliseconds since epoch as int64, and updates to int64, as Jeffrey told, are thread-safe. We can't be so sure about time_point implementation (although it is likely that storing as time_point does exactly that, so it is likely we are not even losing anything). Storing milliseconds was the intent of this PR.